### PR TITLE
Split view: Ensure nav bar is visible when popping to root.

### LINKED ISSUE #6670: Reader: Top bar can be lost in navigation
### Expected behavior
For the top navigation bar to be accessible

### Actual behavior
It can be lost, never to return.

### Steps to reproduce the behavior
1) Open a post in the Reader
2) Scroll down do cause the top bar to hide
3) Swipe from the left of the screen to open the Reader menu
4) Pull down to refresh
5) Go back to the post view and look frantically for the top nav bar

It's not shown in the video, but even navigating to other sections of the app and back doesn't restore the bar

https://cloudup.com/c40oStzxLua

##### Tested on iPhone 6, iOS 10.2.1, WPiOS v6.9.0.20170207

cc: @rachelmcr for the report

### PR DESCRIPTION
**Pull request by frosty**
Fixes #6670.

Note that the reproduction steps are slightly incorrect in the original issue. See 'to test' below for replication steps.

I wanted to fix this in the Reader detail VC, as that's the one that hides the bar in the first place. But it turns out that its `navigationController` gets set to `nil` as soon as `popToRootViewController` is called. I also could've set it in the navigation controller delegate, but I didn't want to presume to update the bars for any and every navigation controller. 

So I've opted to just set it in the method that takes care of popping to the root when you tap on the currently-selected tab bar item.

To test:

1. Open a post in the Reader
2. Scroll down to cause the top bar to hide
3. **Tap the Reader tab bar item to pop back to the Reader menu**
4. Navigation bar has gone.

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/ViewRelated/System/WPSplitViewController.swift [hunk 1]
```diff
@@ -370,6 +370,8 @@ class WPSplitViewController: UISplitViewController {
             } else {
                 navigationController.scrollContentToTopAnimated(animated)
             }
+
+            navigationController.setNavigationBarHidden(false, animated: animated)
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/System/WPSplitViewController.swift]: Hmm... would it be better to move this into the if clause so its only performed when `popToRootViewController` is called?

Comment (aerych) [inline WordPress/Classes/ViewRelated/System/WPSplitViewController.swift]: Should we add a comment about why this is needed for our future selves?

**Comment by aerych** [comment]
The patch works great. :)  I just had a couple of questions.

File: WordPress/Classes/ViewRelated/System/WPSplitViewController.swift (hunk 1 above)
Comment (frosty) [inline WordPress/Classes/ViewRelated/System/WPSplitViewController.swift]: Yes it would, yes we should 👍

I'll update it – thanks!

Comment (frosty) [inline WordPress/Classes/ViewRelated/System/WPSplitViewController.swift]: Thanks @aerych! Just pushed up a tweak. Look okay?

Comment (aerych) [inline WordPress/Classes/ViewRelated/System/WPSplitViewController.swift]: Shiny!

**Pull request merged**